spec/controllers/spree/admin/products_controller_spec.rb

```diff
@@ -74,45 +74,30 @@
   end
 
   context "creating a new product" do
-    before { login_as_admin }
+    let(:supplier) { create(:supplier_enterprise) }
+    let(:taxon) { create(:taxon) }
+    let(:shipping_category) { create(:shipping_category) }
+
+    let(:product_attrs) {
+      attributes_for(:product).merge(
```

Comment:
`attributes_for` is one of FactoryBot's creation methods.

Reply:
nice!!! you can use a factory to generate a hash!

Reply:
Yep, perfect fit for creation controller specs IMO.